### PR TITLE
Add mikrus extension

### PR DESCRIPTION
**Pull request by tkowalski29**
### Features

- ability to preview the server list
- possibility of setting separate keys for each server
- possibility of restarting the server
- possibility of "drugging" the server with amphetamine

### How to use

1. add your main server and token in settings
2. for each server that has been included in the list, you can add a new token

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/mikrus/package.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    {
+      "name": "apiKey",
+      "title": "API Key",
+      "description": "API key for communication",
+      "type": "textfield",
+      "required": true
+    },
+    {
+      "name": "defaultServer",
+      "title": "Default server id",
```
Comment (pernielsentikaer) [inline extensions/mikrus/package.json:34]:
```suggestion
      "title": "Default Server ID",
```

File: extensions/mikrus/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+2. for each server that has been included in the list, you can add a new token
+
+# Preferences
+
+All preferences properties list that can be customize through `Raycast Settings > Extensions > Mikr.Us`
+
+| Properties                   | Label                          | Value       | Required | Default    | Description                                                                                          |
+| ---------------------------- | ------------------------------ | ----------- | -------- | ---------- | ---------------------------------------------------------------------------------------------------- |
+| `apiKey`                     | API Key                        | `string`    | `false`  | `empty`    | Your default api key to get lists servers                                                            |
+| `defaultServer`              | Default server id              | `string`    | `false`  | `empty`    | Your defalt id server                                                                                |
```
Comment (pernielsentikaer) [inline extensions/mikrus/README.md:38]:
```suggestion
| `defaultServer`              | Default Server ID              | `string`    | `false`  | `empty`    | Your defalt id server                                                                                |
```

File: extensions/mikrus/src/server.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      throttle={false}
+      selectedItemId={selectedServerId || undefined}
+      onSelectionChange={(id) => setSelectedServerId(id)}
+      searchBarPlaceholder="Search server..."
+      searchText={searchText}
+      onSearchTextChange={setSearchText}
+      actions={getActionList}
+    >
+      {collectionsServers.data.length === 0 ? (
+        <List.EmptyView title="No servers" description="Reload servers" icon={Icon.Stars} />
```
Comment (pernielsentikaer) [inline extensions/mikrus/src/server.tsx:63]:
```suggestion
        <List.EmptyView title="No Servers" description="Reload servers by pressing enter" icon={Icon.Stars} />
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Some actions in the ActionPanel is missing icons

- Do you mind sharing a screencast so I can see the flow since I don't have an account 🙂

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by tkowalski29** [comment]
@pernielsentikaer video: https://streamable.com/4e0x5c?src=player-page-share

**Comment by mil3na** [comment]
Hi!

Thanks for the video. Some icons on the Action Panel are still missing, as Per noted. Also, I noticed that, when you edit the server, the key needs to be entered again. Is that the case? What if the user makes no change? Will the key be erased?

**Comment by tkowalski29** [comment]
Icons added, now everything should have icons.

Regarding the API key. The API is a bit unfortunate because it requires adding a separate key for each server. But you can download the server list with any key.

I solved it in such a way that we provide the key and default server (I added a condition that if the server matches the one that is default, the key does not need to be provided again). But additionally, it gives the possibility of uploading separate keys for separate servers.

Now there is no longer the effect that the key for the default server is not filled in.

**Comment by mil3na** [comment]
My concern here is that, when you click on "change", on the Action Panel, it seems that the key value is empty and I wonder if the user presses enter without changing anything if the key gets erased or not. It's not clear from your video. It is not good UX if the user needs to enter it if there was no change.

**Comment by tkowalski29** [comment]
@mil3na

Once again, because I see confusion, there is currently no problem with the key and you don't need to enter it twice. Below is a new video.

https://streamable.com/rqaykx

**Review by mil3na — APPROVED** [review]
Got it! Thanks for the video.
You should be good to go!
Cheers!

**Pull request merged**